### ISSUE TITLE
Predictor objects do not serialize output transforms

### ISSUE THREAD

**Issue by canerturkmen** [awslabs/gluon-ts#1469]
## Description
Representable predictor objects do not serialize/deserialize the output transformation. See [here](https://github.com/awslabs/gluon-ts/blob/da8442a8dd7d4e6b0251e7d44a3be39419bee131/src/gluonts/mx/model/predictor.py#L206). This prevents consistent serde as well as the ability to reuse existing predictor objects. Also see the discussion in #1458 .

**Comment by lostella** [comment]
Would this be just as easy as it's done [here](https://github.com/awslabs/gluon-ts/blob/da8442a8dd7d4e6b0251e7d44a3be39419bee131/src/gluonts/mx/model/predictor.py#L202-L204)?

**Comment by canerturkmen** [comment]
that doesn't appear to be the case. In general output transforms are just `Callable`s. For a simple function I get: 
```
RuntimeError: Cannot serialize type builtins.function.
```

**Comment by lostella** [comment]
Yeah that’s an issue of course. In my opinion it should be fine to use whatever callable objects wherever they fit, as long as you’re willing to pay the price that it may not be serializable nicely. We could use for output transforms similar things like we use for the input, right? And have serialization rely on the `@validated` decorator

**Comment by canerturkmen** [comment]
`OutputTransform` is currently an alias for a certain `Callable` interface. It could become a serializable object of its own although seeing as it doesn't look like it's being used much I'm not sure if it's worth the effort.

**Comment by lostella** [comment]
> It could become a serializable object of its own although seeing as it doesn't look like it's being used much I'm not sure if it's worth the effort.

What I mean is that one does not need to do much, in order to provide an object compatible with `dump_json`: just extend `OutputTransform` (to make type checking happy) and decorate the constructor:

```python
In [11]: class MyOutputTransform(OutputTransform):
    ...:     @validated()
    ...:     def __init__(self, coeff: float):
    ...:         self.coeff = coeff
    ...:     def __call__(self, entry, x):
    ...:         return entry["target"] + coeff*x
    ...:

In [12]:

In [12]:

In [12]: output_transform = MyOutputTransform(42.0)

In [13]: encode(output_transform)
Out[13]:
{'__kind__': <Kind.Instance: 'instance'>,
 'class': '__main__.MyOutputTransform',
 'args': [],
 'kwargs': {'coeff': 42.0}}

In [14]: decode(encode(output_transform))
Out[14]: __main__.MyOutputTransform(coeff=42.0)
```

(These `encode`/`decode` functions are what is used internally for serde from/to json)
